src/components/modals/LibraryCardFormModal.tsx

```diff
@@ -26,71 +26,57 @@ interface FormFieldRowProps {
 
 export interface LibraryCardFormModalProps {
   showModal: boolean
-  createOnClick: (cards: LibraryCard[]) => Promise<void>
+  addButton: (cards: LibraryCard[]) => Promise<void>
   closeModal: () => void
   users: UserOption[]
 }
 
-interface FilterOptions {
-  searchTerm?: string
-  input?: string
-  action: string
-}
-
 const FormFieldRow: React.FC<FormFieldRowProps> = (props) => {
   const { selectedUsers, dropdownOptions, updateUsers } = props
 
-  const cardlessOptions = dropdownOptions.filter(option => isNil(option.libraryCard))
-  const [filteredDropdown, setFilteredDropdown] = useState<UserOption[]>(cardlessOptions)
+  // Represents users that do not already have library cards
+  const cardlessUserOptions = dropdownOptions.filter(option => isNil(option.libraryCard))
 
-  // filter function for users dropdown
-  const userListFilter = ({ searchTerm, input }: FilterOptions) => {
-    const term = searchTerm ?? input ?? ''
-    let filteredCopy: UserOption[]
+  // Filter function for auto-completing user dropdown
+  const filterUserOptions = (term: string) => {
+    return cardlessUserOptions.filter((user) => {
+      const filterTarget = (user.displayName + ' ' + user.email).toLowerCase()
+      return filterTarget.includes(term.toLowerCase())
+    })
+  }
 
-    if (isEmpty(term)) {
-      filteredCopy = dropdownOptions
-    }
-    else {
-      const copiedDropdown = cloneDeep(filteredDropdown)
-      filteredCopy = copiedDropdown.filter((user) => {
-        const userNameFilter = !isEmpty(user.displayName) ? includes(term)(user.displayName) : false
-        const emailFilter = !isEmpty(user.email) ? includes(term)(user.email) : false
-        return userNameFilter ?? emailFilter
-      })
-    }
-    setFilteredDropdown(filteredCopy)
+  const loadOptions = (inputValue: string, callback: (options: UserOption[]) => void) => {
+    setTimeout(() => {
+      callback(filterUserOptions(inputValue))
+    }, 0)
```

Comment:
See [AsyncSelect documentation](https://react-select.com/async#loading-asynchronously) for why this is structured this way